sources/csv_reader/__init__.py

```diff
@@ -0,0 +1,64 @@
+"""
+Source, which uses the `filesystem` source and DuckDB
+to extract CSV files data from the given locations.
+"""
+from typing import Iterable, List
+
+import duckdb
+import pendulum
+
+import dlt
+from dlt.common.typing import TDataItem, TAnyDateTime
+from filesystem import filesystem
+from fsspec.implementations.local import LocalFileSystem
+
+from .helpers import add_columns
+
+
+@dlt.resource
+def read_location(files):
+    """A resource to extract data from the given CSV files.
+
+    Args:
+        files (List[FileItem]): A list of files to read.
+
+    Returns:
+        Iterable[TDataItem]: Data items, read from the given CSV files.
+    """
+    state = dlt.current.resource_state()
+    start_from = state.setdefault("last_modified", pendulum.datetime(1970, 1, 1))
+
+    results = []
+    for file in files:
+        if file["modification_date"] <= start_from:
+            continue
+
+        file_res = duckdb.sql(f"""SELECT * FROM read_csv_auto('{file["file_url"]}')""")
+        results += add_columns(file_res.columns, file_res.fetchall())
+
+        state["last_modified"] = max(file["modification_date"], state["last_modified"])
+
+    yield results
+
+
+@dlt.source
+def csv_reader(bucket: str, globs: List[str] = ("*",)) -> Iterable[TDataItem]:
```

Comment:
The user can pass several `globs`. By default the most non-specific glob is used.